vms/platformvm/state/state.go

```diff
@@ -1672,10 +1672,17 @@ func (s *state) initValidatorSets() error {
 }
 
 func (s *state) write(updateValidators bool, height uint64) error {
+	weightDiffs, blsKeyDiffs, err := s.processCurrentStakers()
+	if err != nil {
+		return err
+	}
```

Comment:
this is the part where stakers-related content is scanned to extract specific information to be stored in different parts of the pchain state